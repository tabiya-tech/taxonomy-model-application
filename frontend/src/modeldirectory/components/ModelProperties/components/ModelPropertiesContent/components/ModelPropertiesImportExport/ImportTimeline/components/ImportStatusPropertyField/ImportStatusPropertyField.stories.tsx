import { Meta, StoryObj } from "@storybook/react";
import ImportStatusPropertyField, { uniqueId } from "./ImportStatusPropertyField";
import { ImportProcessStateEnums } from "api-specifications/importProcessState/enums";
import { v4 as id } from "uuid";

const meta: Meta<typeof ImportStatusPropertyField> = {
  title: "ModelDirectory/ModelProperties/ModelPropertiesContentPanels/ModelPropertiesImportExport/ImportTimeline/ImportStatusPropertyField",
  component: ImportStatusPropertyField,
  tags: ["autodocs"],
  args: {
    importProcessState: {
      id: id(),
      status: ImportProcessStateEnums.Status.PENDING,
      result: {
        errored: false,
        parsingWarnings: false,
        parsingErrors: false
      }
    },
    fieldId: uniqueId
  },
};

export default meta;

type Story = StoryObj<typeof ImportStatusPropertyField>;

export const Pending: Story = {
  args: {
    importProcessState: {
      id: id(),
      status: ImportProcessStateEnums.Status.PENDING,
      result: {
        errored: false,
        parsingWarnings: false,
        parsingErrors: false
      }
    },
    fieldId: uniqueId
  }
}

export const Running: Story = {
  args: {
    importProcessState: {
      id: id(),
      status: ImportProcessStateEnums.Status.RUNNING,
      result: {
        errored: false,
        parsingWarnings: false,
        parsingErrors: false
      }
    },
    fieldId: uniqueId
  }
}

export const Completed: Story = {
  args: {
    importProcessState: {
      id: id(),
      status: ImportProcessStateEnums.Status.COMPLETED,
      result: {
        errored: false,
        parsingWarnings: false,
        parsingErrors: false
      }
    },
    fieldId: uniqueId
  }
}

export const CompletedWithCriticalErrors: Story = {
  args: {
    importProcessState: {
      id: id(),
      status: ImportProcessStateEnums.Status.COMPLETED,
      result: {
        errored: true,
        parsingWarnings: false,
        parsingErrors: false
      }
    },
    fieldId: uniqueId
  }
}


export const CompletedWithParsingErrors: Story = {
  args: {
    importProcessState: {
      id: id(),
      status: ImportProcessStateEnums.Status.COMPLETED,
      result: {
        errored: false,
        parsingWarnings: false,
        parsingErrors: true
      }
    },
    fieldId: uniqueId
  }
}

export const CompletedWithParsingWarnings: Story = {
  args: {
    importProcessState: {
      id: id(),
      status: ImportProcessStateEnums.Status.COMPLETED,
      result: {
        errored: false,
        parsingWarnings: true,
        parsingErrors: false
      }
    },
    fieldId: uniqueId
  }
}

export const CompletedWithParsingWarningsAndParsingErrors: Story = {
  args: {
    importProcessState: {
      id: id(),
      status: ImportProcessStateEnums.Status.COMPLETED,
      result: {
        errored: false,
        parsingWarnings: true,
        parsingErrors: true
      }
    },
    fieldId: uniqueId
  }
}
