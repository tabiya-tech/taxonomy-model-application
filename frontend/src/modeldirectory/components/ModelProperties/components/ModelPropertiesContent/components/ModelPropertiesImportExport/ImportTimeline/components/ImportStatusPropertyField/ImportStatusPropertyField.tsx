import React, { PropsWithChildren } from "react";
import Box from "@mui/material/Box";
import HelpTip from "src/theme/HelpTip/HelpTip";
import { Typography, useTheme } from "@mui/material";
import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import ImportProcessStateEnums from "api-specifications/importProcessState";
import ImportProcessStateIcon from "src/modeldirectory/components/ImportProcessStateIcon/ImportProcessStateIcon";
import PropertyFieldLayout from "src/theme/PropertyFieldLayout/PropertyFieldLayout";

interface ImportStatusPropertyFieldProps {
  importProcessState: ModelInfoTypes.ImportProcessState;
  fieldId: string;
  dataTestId?: string;
}

export const uniqueId = "d321d7cc-6396-43c8-bec9-70a04fdb3607";

export const DATA_TEST_ID = {
  PROCCESS_STATUS_FIELD: `process-status-field-${uniqueId}`,
  PROCCESS_STATUS_ICON: `process-status-icon-${uniqueId}`,
  MESSAGE: `message-${uniqueId}`,
  HELP_TIP_SUCCESS: `help-tip-success-${uniqueId}`,
  HELP_TIP_ERROR: `help-tip-error-${uniqueId}`,
  HELP_TIP_PARSING_WARNING: `help-tip-parsing-warning-${uniqueId}`,
  HELP_TIP_PARSING_ERROR: `help-tip-parsing-error-${uniqueId}`,
  HELP_TIP_RUNNING: `help-tip-running-${uniqueId}`,
  HELP_TIP_PENDING: `help-tip-pending-${uniqueId}`,
};

export const HELP_TIP_TEXT = {
  PENDING: "The import process has not yet started.",
  RUNNING: "The import process is running.",
  SUCCESS: "The import process has completed successfully.",
  ERROR:
    "The import process could not start or complete due to a technical error. Please try again later or contact us if the problem persists.",
  PARSING_ERROR:
    "The csv files have inconsistencies that could not be resolved. The model was created but is not consistent. Please review the csv files and try again.",
  PARSING_WARNING:
    "One or more rows or some field could not be imported, but the model was still created and is consistent. Please review the csv files and try again.",
};

const Bold = ({ children }: PropsWithChildren) => (
  <Box component="span" fontWeight="bold">
    {children}
  </Box>
);


const Message = (props: Readonly<{ importProcessState: ModelInfoTypes.ImportProcessState }>) => {
  const { status, result } = props.importProcessState;
  switch (status) {
    case ImportProcessStateEnums.Enums.Status.PENDING:
      return (
        <Typography data-testid={DATA_TEST_ID.MESSAGE}>
          Pending...
          <HelpTip data-testid={DATA_TEST_ID.HELP_TIP_PENDING}>{HELP_TIP_TEXT.PENDING}</HelpTip>
        </Typography>
      )

    case ImportProcessStateEnums.Enums.Status.RUNNING:
      return (
        <Typography data-testid={DATA_TEST_ID.MESSAGE}>
          Running...
          <HelpTip data-testid={DATA_TEST_ID.HELP_TIP_RUNNING}>{HELP_TIP_TEXT.RUNNING}</HelpTip>
        </Typography>
      )

    case ImportProcessStateEnums.Enums.Status.COMPLETED:
      if (!result.errored && !result.parsingErrors && !result.parsingWarnings) {
        return (
          <Typography data-testid={DATA_TEST_ID.MESSAGE}>
            Completed successfully
            <HelpTip data-testid={DATA_TEST_ID.HELP_TIP_SUCCESS}>{HELP_TIP_TEXT.SUCCESS}</HelpTip>
          </Typography>
        )
      }

      if (result.errored && !result.parsingErrors && !result.parsingWarnings) {
        return (
          <Typography data-testid={DATA_TEST_ID.MESSAGE}>
            Completed with <Bold>critical errors</Bold>
            <HelpTip data-testid={DATA_TEST_ID.HELP_TIP_ERROR}>{HELP_TIP_TEXT.ERROR}</HelpTip>
          </Typography>
        )
      }
      if (!result.errored && result.parsingErrors && !result.parsingWarnings) {
        return (
          <Typography data-testid={DATA_TEST_ID.MESSAGE}>
            Completed with <Bold>parsing errors</Bold>
            <HelpTip data-testid={DATA_TEST_ID.HELP_TIP_PARSING_ERROR}>{HELP_TIP_TEXT.PARSING_ERROR}</HelpTip>
          </Typography>
        )
      }
      if (!result.errored && !result.parsingErrors && result.parsingWarnings) {
        return (
          <Typography data-testid={DATA_TEST_ID.MESSAGE}>
            Completed with <Bold>parsing warnings</Bold>
            <HelpTip data-testid={DATA_TEST_ID.HELP_TIP_PARSING_WARNING}>{HELP_TIP_TEXT.PARSING_WARNING}</HelpTip>
          </Typography>
        )
      }
      if (!result.errored && result.parsingErrors && result.parsingWarnings) {
        return (
          <Typography data-testid={DATA_TEST_ID.MESSAGE}>
            <span>Completed with</span>
            <Bold> parsing errors</Bold>
            <HelpTip data-testid={DATA_TEST_ID.HELP_TIP_PARSING_WARNING}>{HELP_TIP_TEXT.PARSING_WARNING}</HelpTip>
            and <Bold>parsing warnings</Bold>
            <HelpTip data-testid={DATA_TEST_ID.HELP_TIP_PARSING_WARNING}>{HELP_TIP_TEXT.PARSING_WARNING}</HelpTip>
          </Typography>
        )
      }
      return <Typography data-testid={DATA_TEST_ID.MESSAGE}>Completed with unexpected results</Typography>;
  }
};

/**
 * ImportStatusPropertyField is responsible for showing the import status
 * @param props
 * @constructor
 */

const ImportStatusPropertyField: React.FC<ImportStatusPropertyFieldProps> = (
  props: Readonly<ImportStatusPropertyFieldProps>
) => {
  const theme = useTheme();

  return (
    <PropertyFieldLayout title="Status" test-dataId={DATA_TEST_ID.PROCCESS_STATUS_FIELD} fieldId={props.fieldId}>
      <Box
        display="flex"
        flexDirection="row"
        alignItems="center"
        gap={theme.spacing(1)}
        data-testid={props["dataTestId"] || DATA_TEST_ID.PROCCESS_STATUS_FIELD}
      >
        <ImportProcessStateIcon
          test-dataId={DATA_TEST_ID.PROCCESS_STATUS_ICON}
          importProcessState={props.importProcessState}
        />
        <Message importProcessState={props.importProcessState} />
      </Box>
    </PropertyFieldLayout>
  );
};

export default ImportStatusPropertyField;
