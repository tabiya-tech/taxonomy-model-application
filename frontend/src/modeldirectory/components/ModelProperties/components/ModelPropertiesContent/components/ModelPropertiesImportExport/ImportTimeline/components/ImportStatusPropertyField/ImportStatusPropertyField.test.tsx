//mute chatty console
import "src/_test_utilities/consoleMock";

import React from "react";
import { v4 as uuid } from "uuid"
import "@testing-library/jest-dom";
import HelpTip from "src/theme/HelpTip/HelpTip";
import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import { render, screen } from "src/_test_utilities/test-utils";
import * as States from "./ImportStatusPropertyField.stories";
import ImportProcessStateEnums from "api-specifications/importProcessState";
import ImportStatusPropertyField, {DATA_TEST_ID, HELP_TIP_TEXT} from "./ImportStatusPropertyField";
import { DATA_TEST_ID as ICON_DATA_TEST_ID } from "src/modeldirectory/components/ImportProcessStateIcon/ImportProcessStateIcon";

import ImportProcessState = ModelInfoTypes.ImportProcessState;

const cases = [
  {
    name: "Pending",
    state: States.Pending.args?.importProcessState,
    icon: ICON_DATA_TEST_ID.ICON_STATUS_PENDING,
    checkedMessages: ["Pending..."],
    helpTipMessages: [HELP_TIP_TEXT.PENDING]
  },
  {
    name: "Running",
    state: States.Running.args?.importProcessState,
    icon: ICON_DATA_TEST_ID.ICON_STATUS_RUNNING,
    checkedMessages: ["Running..."],
    helpTipMessages: [HELP_TIP_TEXT.RUNNING]
  },
  {
    name: "Completed",
    state: States.Completed.args?.importProcessState,
    icon: ICON_DATA_TEST_ID.ICON_STATUS_SUCCESS,
    checkedMessages: ["Completed successfully"],
    helpTipMessages: [HELP_TIP_TEXT.SUCCESS]
  },
  {
    name: "Completed With Critical Errors",
    state: States.CompletedWithCriticalErrors.args?.importProcessState,
    icon: ICON_DATA_TEST_ID.ICON_STATUS_FAILED,
    checkedMessages: ["Completed with", "critical errors"],
    helpTipMessages: [HELP_TIP_TEXT.ERROR]
  },
  {
    name: "Completed With Parsing Errors",
    state: States.CompletedWithParsingErrors.args?.importProcessState,
    icon: ICON_DATA_TEST_ID.ICON_STATUS_FAILED,
    checkedMessages: ["Completed with", "parsing errors"],
    helpTipMessages: [HELP_TIP_TEXT.PARSING_ERROR]
  },
  {
    name: "Completed With Parsing Warnings",
    state: States.CompletedWithParsingWarnings.args?.importProcessState,
    icon: ICON_DATA_TEST_ID.ICON_STATUS_FAILED,
    checkedMessages: ["Completed with", "parsing warnings"],
    helpTipMessages: [HELP_TIP_TEXT.PARSING_WARNING]
  },
  {
    name: "Completed With Parsing Errors and Parsing Warnings",
    state: States.CompletedWithParsingWarningsAndParsingErrors.args?.importProcessState,
    icon: ICON_DATA_TEST_ID.ICON_STATUS_FAILED,
    checkedMessages: ["Completed with", "parsing errors", "parsing warnings"],
    helpTipMessages: [HELP_TIP_TEXT.PARSING_WARNING, HELP_TIP_TEXT.PARSING_ERROR]
  },
]

jest.mock("src/theme/HelpTip/HelpTip", () => ({
  __esModule: true,
  default: jest.fn().mockImplementation(() => {
    return <></>;
  }),
}));

describe('ImportStatusPropertyField', () => {
  let id = uuid();

  test.each(cases)(`Should Render correctly for $name`, (testCase) => {
      // GIVEN the ImportStatusPropertyField component is rendered with a specific import state.
      const importProcessState: ImportProcessState = {
        id: uuid(),
        status: testCase.state?.status!,
        result: testCase.state?.result!
      };

      // WHEN the component mounts
      render(<ImportStatusPropertyField importProcessState={importProcessState} fieldId={id} />);

      // THEN it should display the correct status icon if not unknown
      const statusIcon = screen.getByTestId(testCase.icon)
      expect(statusIcon).toBeInTheDocument()

      // AND the message should be correct
      testCase.checkedMessages.forEach((message) => {
        expect(screen.getAllByText(message).length).toBeGreaterThanOrEqual(1)
      })

      // AND no errors or warnings should have occurred
      expect(console.error).not.toHaveBeenCalled();
      expect(console.warn).not.toHaveBeenCalled();

      // AND Help Tip Messages should be present
      testCase.helpTipMessages.forEach((message) => {
        expect(HelpTip).toHaveBeenCalledWith({
          "data-testid": expect.any(String),
          children: message
        }, {})
      })

      // AND it should match the snapshot
      expect(screen.getByTestId(DATA_TEST_ID.PROCCESS_STATUS_FIELD)).toMatchSnapshot()
    })

  test("Should return Unknown Status on Invalid case", () => {
    // GIVEN the ImportStatusPropertyField component is rendered with completed with a case that won't exist
    const importProcessState: ImportProcessState = {
      id: uuid(),
      status: ImportProcessStateEnums.Enums.Status.COMPLETED,
      result: {
        errored: true,
        parsingErrors: true,
        parsingWarnings: true
      }
    };

    // WHEN the component mounts
    render(<ImportStatusPropertyField importProcessState={importProcessState} fieldId={id} />);

    // THEN it should return unknown status
    expect(screen.getByText("Completed with unexpected results")).toBeInTheDocument()
  })
})
