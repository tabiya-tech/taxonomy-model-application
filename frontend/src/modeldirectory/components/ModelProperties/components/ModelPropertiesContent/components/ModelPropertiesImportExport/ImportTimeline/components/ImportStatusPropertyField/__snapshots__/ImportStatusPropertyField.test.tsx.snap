// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportStatusPropertyField Should Render correctly for % 1`] = `
<div
  class="MuiBox-root css-1t5r8n4"
  data-testid="process-status-field-d321d7cc-6396-43c8-bec9-70a04fdb3607"
>
  <svg
    class="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-14fk8m9-MuiSvgIcon-root"
    data-testid="icon-status-pending-bae86ed9-33bf-4492-a0e2-f9c8bd112bae"
    focusable="false"
    role="img"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm4.2 14.2L11 13V7h1.5v5.2l4.5 2.7-.8 1.3z"
    />
    <title>
      Pending
    </title>
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-3ahvbo-MuiTypography-root"
    data-testid="message-d321d7cc-6396-43c8-bec9-70a04fdb3607"
  >
    Pending...
  </p>
</div>
`;

exports[`ImportStatusPropertyField Should Render correctly for % 2`] = `
<div
  class="MuiBox-root css-1t5r8n4"
  data-testid="process-status-field-d321d7cc-6396-43c8-bec9-70a04fdb3607"
>
  <svg
    class="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-1wn4sen-MuiSvgIcon-root"
    data-testid="icon-status-running-bae86ed9-33bf-4492-a0e2-f9c8bd112bae"
    focusable="false"
    role="img"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2z"
    />
    <title>
      Running
    </title>
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-3ahvbo-MuiTypography-root"
    data-testid="message-d321d7cc-6396-43c8-bec9-70a04fdb3607"
  >
    Running...
  </p>
</div>
`;

exports[`ImportStatusPropertyField Should Render correctly for % 3`] = `
<div
  class="MuiBox-root css-1t5r8n4"
  data-testid="process-status-field-d321d7cc-6396-43c8-bec9-70a04fdb3607"
>
  <svg
    class="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeMedium css-1x9cg3q-MuiSvgIcon-root"
    data-testid="icon-status-success-bae86ed9-33bf-4492-a0e2-f9c8bd112bae"
    focusable="false"
    role="img"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
    />
    <title>
      Completed succesfully
    </title>
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-3ahvbo-MuiTypography-root"
    data-testid="message-d321d7cc-6396-43c8-bec9-70a04fdb3607"
  >
    Completed successfully
  </p>
</div>
`;

exports[`ImportStatusPropertyField Should Render correctly for % 4`] = `
<div
  class="MuiBox-root css-1t5r8n4"
  data-testid="process-status-field-d321d7cc-6396-43c8-bec9-70a04fdb3607"
>
  <svg
    class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-13bdyef-MuiSvgIcon-root"
    data-testid="icon-status-failed-bae86ed9-33bf-4492-a0e2-f9c8bd112bae"
    focusable="false"
    role="img"
    viewBox="0 0 24 24"
  >
    <path
      d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
    />
    <title>
      Completed with error(s)
    </title>
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-3ahvbo-MuiTypography-root"
    data-testid="message-d321d7cc-6396-43c8-bec9-70a04fdb3607"
  >
    Completed with 
    <span
      class="MuiBox-root css-1vg6q84"
    >
      critical errors
    </span>
  </p>
</div>
`;

exports[`ImportStatusPropertyField Should Render correctly for % 5`] = `
<div
  class="MuiBox-root css-1t5r8n4"
  data-testid="process-status-field-d321d7cc-6396-43c8-bec9-70a04fdb3607"
>
  <svg
    class="MuiSvgIcon-root MuiSvgIcon-colorWarning MuiSvgIcon-fontSizeMedium css-h6esa3-MuiSvgIcon-root"
    data-testid="icon-status-failed-bae86ed9-33bf-4492-a0e2-f9c8bd112bae"
    focusable="false"
    role="img"
    viewBox="0 0 24 24"
  >
    <path
      d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
    />
    <title>
      Completed with parsing errors(s)
    </title>
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-3ahvbo-MuiTypography-root"
    data-testid="message-d321d7cc-6396-43c8-bec9-70a04fdb3607"
  >
    Completed with 
    <span
      class="MuiBox-root css-1vg6q84"
    >
      parsing errors
    </span>
  </p>
</div>
`;

exports[`ImportStatusPropertyField Should Render correctly for % 6`] = `
<div
  class="MuiBox-root css-1t5r8n4"
  data-testid="process-status-field-d321d7cc-6396-43c8-bec9-70a04fdb3607"
>
  <svg
    class="MuiSvgIcon-root MuiSvgIcon-colorWarning MuiSvgIcon-fontSizeMedium css-h6esa3-MuiSvgIcon-root"
    data-testid="icon-status-failed-bae86ed9-33bf-4492-a0e2-f9c8bd112bae"
    focusable="false"
    role="img"
    viewBox="0 0 24 24"
  >
    <path
      d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
    />
    <title>
      Completed with parsing warning(s)
    </title>
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-3ahvbo-MuiTypography-root"
    data-testid="message-d321d7cc-6396-43c8-bec9-70a04fdb3607"
  >
    Completed with 
    <span
      class="MuiBox-root css-1vg6q84"
    >
      parsing warnings
    </span>
  </p>
</div>
`;

exports[`ImportStatusPropertyField Should Render correctly for % 7`] = `
<div
  class="MuiBox-root css-1t5r8n4"
  data-testid="process-status-field-d321d7cc-6396-43c8-bec9-70a04fdb3607"
>
  <svg
    class="MuiSvgIcon-root MuiSvgIcon-colorWarning MuiSvgIcon-fontSizeMedium css-h6esa3-MuiSvgIcon-root"
    data-testid="icon-status-failed-bae86ed9-33bf-4492-a0e2-f9c8bd112bae"
    focusable="false"
    role="img"
    viewBox="0 0 24 24"
  >
    <path
      d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
    />
    <title>
      Completed with parsing errors(s)
    </title>
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-3ahvbo-MuiTypography-root"
    data-testid="message-d321d7cc-6396-43c8-bec9-70a04fdb3607"
  >
    <span>
      Completed with
    </span>
    <span
      class="MuiBox-root css-1vg6q84"
    >
       parsing errors
    </span>
    and 
    <span
      class="MuiBox-root css-1vg6q84"
    >
      parsing warnings
    </span>
  </p>
</div>
`;

exports[`ImportStatusPropertyField Should Render correctly for Completed 1`] = `
<div
  class="MuiBox-root css-1t5r8n4"
  data-testid="process-status-field-d321d7cc-6396-43c8-bec9-70a04fdb3607"
>
  <svg
    class="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeMedium css-1x9cg3q-MuiSvgIcon-root"
    data-testid="icon-status-success-bae86ed9-33bf-4492-a0e2-f9c8bd112bae"
    focusable="false"
    role="img"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
    />
    <title>
      Completed succesfully
    </title>
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-3ahvbo-MuiTypography-root"
    data-testid="message-d321d7cc-6396-43c8-bec9-70a04fdb3607"
  >
    Completed successfully
  </p>
</div>
`;

exports[`ImportStatusPropertyField Should Render correctly for Completed With Critical Errors 1`] = `
<div
  class="MuiBox-root css-1t5r8n4"
  data-testid="process-status-field-d321d7cc-6396-43c8-bec9-70a04fdb3607"
>
  <svg
    class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-13bdyef-MuiSvgIcon-root"
    data-testid="icon-status-failed-bae86ed9-33bf-4492-a0e2-f9c8bd112bae"
    focusable="false"
    role="img"
    viewBox="0 0 24 24"
  >
    <path
      d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
    />
    <title>
      Completed with error(s)
    </title>
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-3ahvbo-MuiTypography-root"
    data-testid="message-d321d7cc-6396-43c8-bec9-70a04fdb3607"
  >
    Completed with 
    <span
      class="MuiBox-root css-1vg6q84"
    >
      critical errors
    </span>
  </p>
</div>
`;

exports[`ImportStatusPropertyField Should Render correctly for Completed With Parsing Errors 1`] = `
<div
  class="MuiBox-root css-1t5r8n4"
  data-testid="process-status-field-d321d7cc-6396-43c8-bec9-70a04fdb3607"
>
  <svg
    class="MuiSvgIcon-root MuiSvgIcon-colorWarning MuiSvgIcon-fontSizeMedium css-h6esa3-MuiSvgIcon-root"
    data-testid="icon-status-failed-bae86ed9-33bf-4492-a0e2-f9c8bd112bae"
    focusable="false"
    role="img"
    viewBox="0 0 24 24"
  >
    <path
      d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
    />
    <title>
      Completed with parsing errors(s)
    </title>
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-3ahvbo-MuiTypography-root"
    data-testid="message-d321d7cc-6396-43c8-bec9-70a04fdb3607"
  >
    Completed with 
    <span
      class="MuiBox-root css-1vg6q84"
    >
      parsing errors
    </span>
  </p>
</div>
`;

exports[`ImportStatusPropertyField Should Render correctly for Completed With Parsing Errors and Parsing Warnings 1`] = `
<div
  class="MuiBox-root css-1t5r8n4"
  data-testid="process-status-field-d321d7cc-6396-43c8-bec9-70a04fdb3607"
>
  <svg
    class="MuiSvgIcon-root MuiSvgIcon-colorWarning MuiSvgIcon-fontSizeMedium css-h6esa3-MuiSvgIcon-root"
    data-testid="icon-status-failed-bae86ed9-33bf-4492-a0e2-f9c8bd112bae"
    focusable="false"
    role="img"
    viewBox="0 0 24 24"
  >
    <path
      d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
    />
    <title>
      Completed with parsing errors(s)
    </title>
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-3ahvbo-MuiTypography-root"
    data-testid="message-d321d7cc-6396-43c8-bec9-70a04fdb3607"
  >
    <span>
      Completed with
    </span>
    <span
      class="MuiBox-root css-1vg6q84"
    >
       parsing errors
    </span>
    and 
    <span
      class="MuiBox-root css-1vg6q84"
    >
      parsing warnings
    </span>
  </p>
</div>
`;

exports[`ImportStatusPropertyField Should Render correctly for Completed With Parsing Warnings 1`] = `
<div
  class="MuiBox-root css-1t5r8n4"
  data-testid="process-status-field-d321d7cc-6396-43c8-bec9-70a04fdb3607"
>
  <svg
    class="MuiSvgIcon-root MuiSvgIcon-colorWarning MuiSvgIcon-fontSizeMedium css-h6esa3-MuiSvgIcon-root"
    data-testid="icon-status-failed-bae86ed9-33bf-4492-a0e2-f9c8bd112bae"
    focusable="false"
    role="img"
    viewBox="0 0 24 24"
  >
    <path
      d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
    />
    <title>
      Completed with parsing warning(s)
    </title>
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-3ahvbo-MuiTypography-root"
    data-testid="message-d321d7cc-6396-43c8-bec9-70a04fdb3607"
  >
    Completed with 
    <span
      class="MuiBox-root css-1vg6q84"
    >
      parsing warnings
    </span>
  </p>
</div>
`;

exports[`ImportStatusPropertyField Should Render correctly for Pending 1`] = `
<div
  class="MuiBox-root css-1t5r8n4"
  data-testid="process-status-field-d321d7cc-6396-43c8-bec9-70a04fdb3607"
>
  <svg
    class="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-14fk8m9-MuiSvgIcon-root"
    data-testid="icon-status-pending-bae86ed9-33bf-4492-a0e2-f9c8bd112bae"
    focusable="false"
    role="img"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm4.2 14.2L11 13V7h1.5v5.2l4.5 2.7-.8 1.3z"
    />
    <title>
      Pending
    </title>
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-3ahvbo-MuiTypography-root"
    data-testid="message-d321d7cc-6396-43c8-bec9-70a04fdb3607"
  >
    Pending...
  </p>
</div>
`;

exports[`ImportStatusPropertyField Should Render correctly for Running 1`] = `
<div
  class="MuiBox-root css-1t5r8n4"
  data-testid="process-status-field-d321d7cc-6396-43c8-bec9-70a04fdb3607"
>
  <svg
    class="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-1wn4sen-MuiSvgIcon-root"
    data-testid="icon-status-running-bae86ed9-33bf-4492-a0e2-f9c8bd112bae"
    focusable="false"
    role="img"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2z"
    />
    <title>
      Running
    </title>
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-3ahvbo-MuiTypography-root"
    data-testid="message-d321d7cc-6396-43c8-bec9-70a04fdb3607"
  >
    Running...
  </p>
</div>
`;
